Dev

ValentinPside opened the pull request:
Домашняя работа по навигации и разметке

ValentinPside committed:
hw

ValentinPside committed:
hw2

SeveNChaK commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,32 @@
+        settingsButton.setOnClickListener {
+            val displayIntent = Intent(this, Settings::class.java)
+            startActivity(displayIntent)
+        }
```
⏫ Можно чуток сократить, сделав общую функцию для навигации:
```
private fun navigateTo(clazz: Class<out AppCompatActivity>) {
    val intent = Intent(this, clazz)
    startActivity(intent)
}
```
и вызывать эту функцию в соответствующем месте с нужным классом `navigateTo(НужныйКласс::class.java)`

SeveNChaK commented on app/src/main/res/drawable/search.xml:
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="19dp"
```
⏫ Все иконки не совсем корректно экcпортированы из фигмы. Обращай внимание на иерархию элементов в фигма, когда выбираешь иконку. Как правило, все иконки квадратные и в дизайнах андроида имеют размеры 16, 24, 32.

❌ Выбран не верный уровень вложенности
<img width="332" alt="image" src="https://user-images.githubusercontent.com/22057568/220430031-e818935c-4c12-4d17-a956-57c832a1ba50.png">

✅ Верный уровень вложенности
<img width="389" alt="image" src="https://user-images.githubusercontent.com/22057568/220430146-5a195fa0-b5f8-4f8b-bc3e-26866dd86959.png">

Если правильно достать иконку, то с ней будет легче работать и проще проставлять отступы, так как иконки будут одинаковые по размеру, как и задумывал автор макета. Рекомендую пройтись по всем ресурсам.

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
-        app:layout_constraintTop_toTopOf="parent" />
+        android:id="@+id/title"
+        android:layout_width="134dp"
+        android:layout_height="26dp"
```
⚠️ Обрезается заголовок
<img width="406" alt="image" src="https://user-images.githubusercontent.com/22057568/230741068-6bf095d6-a5a3-4516-b68a-8e2887088fde.png">

В макете
<img width="391" alt="image" src="https://user-images.githubusercontent.com/22057568/230741078-62a13dff-6827-477e-9735-00466d4df7d7.png">

⏫ Тут хватило бы `wrap_content` в рамзерах. Лучше не использовать фиксированный размер для текстовых полей, потому что когда пользователь меняет размер текста в системе, то он увеличивается и в приложении, и может не влезть в фиксированный размер. Поэтому для текстовых полей лучше использовать относительные размеры `wrap_content` или `match_parent` или constrains, если используешь `ConstraintLayout`, а остальное задавать через отступы.

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
-    tools:context=".MainActivity">
+    android:background="@color/main_col"
+    android:orientation="vertical"
+    android:padding="20dp">
```
⚠️ Со всех сторон разный отступ.
Сверху 14
<img width="261" alt="image" src="https://user-images.githubusercontent.com/22057568/230741046-414b2a63-bfda-4d5f-8de7-5dc03e6c087c.png">

Слева и справа отступ 16
<img width="421" alt="image" src="https://user-images.githubusercontent.com/22057568/230741031-ef1fcd2c-6fb7-42d2-8b2c-a1d36d0ccbb9.png">

А снизу 28
<img width="452" alt="image" src="https://user-images.githubusercontent.com/22057568/230741051-bc0fe3d7-8308-483e-9590-bcdffce6a1ee.png">

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
+        android:layout_height="26dp"
+        android:text="@string/title"
+        android:textAppearance="@style/title"
+        android:gravity="top" />
```
⏫ В этом аттрибуте нет необходимости

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_marginTop="20dp"
```
⚠️ Здесь отступ 40
<img width="205" alt="image" src="https://user-images.githubusercontent.com/22057568/230741140-be20d540-2c85-48fd-9d25-25fbdf5812c2.png">

⏫ Кажется, что кнопки и заголовок можно было положить в один общий `LinearLayout`

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
+            android:backgroundTint="@color/but_main_col"
+            android:text="@string/search"
+            android:textColor="@color/black"
+            app:cornerRadius="25dp"
```
⚠️ В ресурсы, так как используется несколько раз
А также в макете скругление 16
<img width="413" alt="image" src="https://user-images.githubusercontent.com/22057568/230741213-ec7d5572-42a8-4939-83ec-71627f256fb6.png">

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
+            app:cornerRadius="25dp"
+            app:icon="@drawable/search"
+            app:iconGravity="textStart"
+            app:iconPadding="1dp"
```
⚠️ Отсутп между иконкой и текстом во всех кнопках равен 8 (и не забудь в ресурсы вынести)
<img width="175" alt="image" src="https://user-images.githubusercontent.com/22057568/230741229-419522cb-3834-4e52-87b9-bdb5856e986f.png">

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,70 @@
+            android:layout_marginTop="0dp"
+            android:layout_weight="1"
+            android:backgroundTint="@color/but_main_col"
+            android:text="@string/search"
```
⚠️ Не задается размер текста. А также текст не должен быть капсом

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,77 @@
+    android:orientation="vertical"
+    tools:context=".Settings" >
+
+    <Button
```
⚠️ В макете тулбар выглядит по другому: единый фон без теней, а также нажиматься должна только стрелочка а не весь тулбар
<img width="422" alt="image" src="https://user-images.githubusercontent.com/22057568/230741374-b8d24b74-56c8-4715-99e8-4015094d14ca.png">

У тебя
<img width="409" alt="image" src="https://user-images.githubusercontent.com/22057568/230741383-5676cfde-fa33-4594-9d30-918dabfcca5e.png">


Рекомендую использовать `Toolbar` view. В него можно положить любую верстку. Это будет примерно так, только подставь свои нужные значения:
```
<androidx.appcompat.widget.Toolbar
    android:id="@+id/settings_toolbar"
    android:layout_width="match_parent"
    android:layout_height="?attr/actionBarSize"
    app:navigationIcon="@drawable/arrow_back_btn"
    android:layout_marginBottom="24dp">

    <TextView
        android:layout_width="wrap_content"
        android:layout_height="match_parent"
        android:text="@string/settings"
        android:textColor="@color/main_text"
        android:textSize="22sp"
        android:textStyle="bold"
        android:gravity="center_vertical" />

</androidx.appcompat.widget.Toolbar>
```
Получается как-то так:
<img width="407" alt="image" src="https://user-images.githubusercontent.com/22057568/223215856-24481053-6d3c-4b2d-9da7-2312389af336.png">

И еще придется дописать в активити в методе `onCreate` немного кода:
```
 override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_settings)
     val toolbar = findViewById<Toolbar>(R.id.settings_toolbar)
     toolbar.setNavigationOnClickListener { onBackPressed() }
 }
```
Нужно найти наш `Toolbar` и сказать, что по клику на иконку, мы делаем действие \"назад\"."

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,77 @@
+        android:id="@+id/switch2"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="40dp"
```
⚠️ Если задать правильный `padding` сверху и снизу (21)
<img width="419" alt="image" src="https://user-images.githubusercontent.com/22057568/230741446-02afb83f-2dab-4e9e-b033-f55c6062fd0d.png">
И поправить тулбар с использванием моей рекомендации, то будет видно, что отступ сверху равен 24
<img width="417" alt="image" src="https://user-images.githubusercontent.com/22057568/230741464-601e527d-2022-4293-8cdf-038659ca3687.png">

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,77 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="40dp"
+        android:layout_marginHorizontal="16dp"
```
⚠️ Отступ слева и справа разные. У всех кнопок слева 16, а справа 12 (у свитча немного другой, но подгоним его под 12, чтобы как у всех)
<img width="141" alt="image" src="https://user-images.githubusercontent.com/22057568/230741583-5f6f69e8-273d-4a9f-bf1e-6493bc615a09.png">
<img width="168" alt="image" src="https://user-images.githubusercontent.com/22057568/230741612-4b06de33-70c8-4e4c-9fd7-8c43a0fee0ec.png">


Для целостности картины рекомендую использовать падинг.

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,77 @@
+        android:layout_height="wrap_content"
+        android:layout_marginTop="40dp"
+        android:layout_marginHorizontal="16dp"
+        android:fontFamily="sans-serif-medium"
```
⚠️ Это добавляет немного жирности, однако в макете он обычный, так что надо убрать

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,77 @@
+        android:fontFamily="sans-serif-medium"
+        android:text="@string/dark_style"
+        android:textAlignment="viewStart"
+        android:textSize="16sp"
```
⚠️ В ресурсы, так как используется несколкько раз

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,77 @@
+        android:textSize="16sp"
+        tools:ignore="UseSwitchCompatOrMaterialXml" />
+
+    <Button
```
⚠️ Тут не понятно, что с отступами, есть какие-то тени, шрифт жирный, цвет иконки не тот.
<img width="474" alt="image" src="https://user-images.githubusercontent.com/22057568/230741751-362f9faf-c7b7-41cb-8798-198c3df425b4.png">

Макет 
<img width="698" alt="image" src="https://user-images.githubusercontent.com/22057568/230741759-66fa7cbd-2b50-48f7-a840-64ae4a751034.png">


Лучше передалть все кнопки здесь как текст с отдельной иконкой. Так ты сможешь настроить все паарметры удобно и страница будет соответствовать макету. Я вижу это как-то так:
```
<LinearLayout
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:paddingStart="16dp"
    android:paddingEnd="12dp">
    <TextView
        android:layout_width="0dp"
        android:layout_height="wrap_content"
        android:layout_marginEnd="8dp"
        android:layout_weight="1"
        android:layout_marginVertical="21dp"
        android:text="@string/share"
        android:textColor="@color/black"
        android:textSize="16sp" />
    <ImageView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_gravity="center_vertical"
        android:tint="#AEAFB4"
        android:src="@drawable/share" />
</LinearLayout>
```
<img width="468" alt="image" src="https://user-images.githubusercontent.com/22057568/230741834-2674a0aa-56a7-4dbc-b13b-ab593c0db12b.png">
И не забудь нужные вещи вынести в ресурсы. А если еще и иконки поправишь по моей рекомендации, то станет еще лучше

SeveNChaK commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="main_but_width">328dp</dimen>
+    <dimen name="main_but_height">220dp</dimen>
```
⏫ Эти размеры больше нигде не используются, можно удалить

SeveNChaK reviewed (commented):
Не надо добавлять к отслеживанию файлы в папке `build`. Они генерируются сами при сборке проекта. Добаь эту папку в `.gitignore`

ValentinPside committed:
hw2

Исправление замечаний

ValentinPside commented:
Внесены изменения для исправления замечаний

ValentinPside committed:
Исправление замечаний

Внесены изменения для исправления замечаний

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,76 @@
-    tools:context=".MainActivity">
+    android:background="@color/main_col"
+    android:orientation="vertical"
+    android:padding="20dp">
```
⚠️ Теперь это лишний отступ, так как ты корректно прописал отступы для всего остального отдельно, это место лишь добавляет пустого пространства.
<img width="391" alt="image" src="https://user-images.githubusercontent.com/22057568/230881793-7e98f593-9dd5-47d7-9242-42f2082b8a8e.png">
Надо убрать это

В макете
<img width="368" alt="image" src="https://user-images.githubusercontent.com/22057568/230881873-ea21bda8-f161-45bf-bed2-bd7777e95bd7.png">

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,76 @@
-        app:layout_constraintTop_toTopOf="parent" />
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="14dp"
+        android:layout_marginBottom="16dp"
```
⚠️ Добавляется лишний отступ снизу. У Первой кнопки и так уже задан нужный отступ в 40

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,109 @@
+        android:paddingLeft="@dimen/setting_but_padding_left"
+        android:paddingRight="@dimen/setting_but_padding_right"
+        android:paddingTop="@dimen/setting_text_padding_top_and_bottom"
+        android:paddingBottom="@dimen/setting_text_padding_top_and_bottom"
```
⚠️ Тут сверху и снизу тоже должно быть 21

ValentinPside committed:
hw3

1)Убран лишний отступ у LinearLayout
2)Убран лишний нижний отступ у TextView
3)Изменён нижний и верхний отступ у Switch на 21dp

ValentinPside commented:
1)Убран лишний отступ у LinearLayout
2)Убран лишний нижний отступ у TextView
3)Изменён нижний и верхний отступ у Switch на 21dp

ValentinPside committed:
Спринт 9

Первая попытка

ValentinPside commented:
Спринт 9
1я попытка

faritowich commented on app/src/main/java/com/example/playlistmaker/Settings.kt:
```diff
@@ -0,0 +1,38 @@
+import android.widget.ImageView
+import androidx.appcompat.widget.Toolbar
+
+class Settings : AppCompatActivity() {
```
Принято давать активити названия, содержащие "Activity", в данном случае SettingsActivity

faritowich commented on app/src/main/java/com/example/playlistmaker/Settings.kt:
```diff
@@ -0,0 +1,38 @@
+            supportIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("valentinmalts@yandex.ru"))
+            supportIntent.putExtra(Intent.EXTRA_SUBJECT, R.string.SettingsSubject)
+            supportIntent.putExtra(Intent.EXTRA_TEXT, R.string.SettingsText)
+            startActivity(supportIntent)
```
Можно ещё использовать функции области видимости, чтобы такие фрагменты кода сокращать до:

```
            Intent(Intent.ACTION_SENDTO).apply {
                data = Uri.parse("mailto:")
                putExtra(Intent.EXTRA_EMAIL, arrayOf("valentinmalts@yandex.ru"))
                putExtra(Intent.EXTRA_SUBJECT, R.string.SettingsSubject)
                putExtra(Intent.EXTRA_TEXT, R.string.SettingsText)
                startActivity(this)
            }

```
Про такие методы почитать можно здесь:
https://kotlinlang.ru/docs/reference/scope-functions.html

faritowich commented on app/src/main/java/com/example/playlistmaker/Search.kt:
```diff
@@ -0,0 +1,65 @@
+    companion object {
+        const val EDIT_TEXT = "EDIT_TEXT"
+    }
+    var searchLineText : String? = null
```
private

faritowich commented on app/src/main/java/com/example/playlistmaker/Search.kt:
```diff
@@ -0,0 +1,65 @@
+
+class Search : AppCompatActivity() {
+    companion object {
+        const val EDIT_TEXT = "EDIT_TEXT"
```
private

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,106 @@
+        <item name="android:paddingTop">16dp</item>
+        <item name="android:paddingBottom">16dp</item>
+        <item name="android:layout_marginLeft">16dp</item>
+        <item name="android:layout_marginRight">16dp</item>
```
Лучше внешние отступы, высоту и ширину, веса не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

faritowich replied:
Лучше внешние отступы, высоту и ширину, веса не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,24 @@
+    <dimen name="setting_but_text_size">16sp</dimen>
+    <dimen name="setting_but_text_end">16dp</dimen>
+    <dimen name="setting_but_text_width">0dp</dimen>
+    <dimen name="setting_but_text_vertical">21dp</dimen>
```
Рекомендуется делать размеры кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented:
Привет! Также хочу обратить внимание на то, что для каждого задания стоит создавать свой ПР, чтобы изменения по разным заданиям не смешивались между собой. После завершения задания надо:
- просто вмержить ПР в основную ветку
-  создать новую ветку от неё же и начать делать следующее задание
- создать новый ПР из новой ветки в главную
- после успешной проверки задания вмержить её
- и т.д.

ValentinPside committed:
9й спринт (2я попытка)

Исправление замечаний

ValentinPside commented:
Исправлены следующие критические замечания:
1)При нажатии на кнопку Очистить на экране поиска клавиатура теперь скрывается
2)Внесены исправления в интенты Написать в поддержку и Поделиться приложением
3)В темном режиме введенный текст на экране поиска чёрный
Дополнительно внесены некоторые изменения из некритических замечаний и взято на заметку, что по успешной проверке нужно вмержить ПР в основную ветку

ValentinPside committed:
9й спринт(3я попытка)

Забыл про манифест, на всякий случай перезалил проект целиком

ValentinPside committed:
3я(полный перезалив)

Дополнительно проверил на своём устройстве и внёс незначительные изменения

ValentinPside commented:
![1](https://user-images.githubusercontent.com/91324342/236567135-525e517e-e15b-42e6-b3ec-fe57aeb1d670.jpg)
![2](https://user-images.githubusercontent.com/91324342/236567138-ad7f2cc7-f2ef-410c-9ba4-621a8e05b025.jpg)
![3](https://user-images.githubusercontent.com/91324342/236567140-1ebca490-df2c-42f7-90be-52e2b07641b6.jpg)

ValentinPside commented:
Приложил скрины с телефона, но что-то с размерами не угадал
